NBitcoin.Altcoins/Stratis.cs

```diff
@@ -0,0 +1,660 @@
+					result = CreateHeadersPayload();
+					return true;
+				}
+
+				return base.TryCreateNew(type, out result);
+			}
+
+			public HeadersPayload CreateHeadersPayload()
+			{
+				return new StratisHeadersPayload();
+			}
+		}
+
+		public class StratisHeadersPayload : HeadersPayload
+		{
+			class BlockHeaderWithTxCount : IBitcoinSerializable
+			{
+				public BlockHeaderWithTxCount()
+				{
+
+				}
+				public BlockHeaderWithTxCount(BlockHeader header)
+				{
+					_Header = header;
+				}
+				internal BlockHeader _Header;
+				#region IBitcoinSerializable Members
+
+				public void ReadWrite(BitcoinStream stream)
+				{
+					stream.ReadWrite(ref _Header);
+					VarInt txCount = new VarInt(0);
+					stream.ReadWrite(ref txCount);
+
+					// Stratis specific addition - unknown usage see Stratis source.
+					stream.ReadWrite(ref txCount);
+				}
+
+				#endregion
+			}
+
+			public override void ReadWriteCore(BitcoinStream stream)
+			{
+				if (stream.Serializing)
+				{
+					var heardersOff = Headers.Select(h => new BlockHeaderWithTxCount(h)).ToList();
+					stream.ReadWrite(ref heardersOff);
+				}
+				else
+				{
+					Headers.Clear();
+					List<BlockHeaderWithTxCount> headersOff = new List<BlockHeaderWithTxCount>();
+					stream.ReadWrite(ref headersOff);
+					Headers.AddRange(headersOff.Select(h => h._Header));
+				}
+			}
+		}
+
+		public class StratisBlockSignature : IBitcoinSerializable
+		{
+			protected bool Equals(StratisBlockSignature other)
+			{
+				return Equals(signature, other.signature);
+			}
+
+			public override bool Equals(object obj)
+			{
+				if (ReferenceEquals(null, obj)) return false;
+				if (ReferenceEquals(this, obj)) return true;
+				if (obj.GetType() != this.GetType()) return false;
+				return Equals((StratisBlockSignature)obj);
+			}
+
+			public override int GetHashCode()
+			{
+				return (signature?.GetHashCode() ?? 0);
+			}
+
+			public StratisBlockSignature()
+			{
+				this.signature = new byte[0];
+			}
+
+			private byte[] signature;
+
+			public byte[] Signature
+			{
+				get
+				{
+					return signature;
+				}
+				set
+				{
+					signature = value;
+				}
+			}
+
+			internal void SetNull()
+			{
+				signature = new byte[0];
+			}
+
+			public bool IsEmpty()
+			{
+				return !this.signature.Any();
+			}
+
+			public static bool operator ==(StratisBlockSignature a, StratisBlockSignature b)
+			{
+				if (System.Object.ReferenceEquals(a, b))
+					return true;
+
+				if (((object)a == null) || ((object)b == null))
+					return false;
+
+				return a.signature.SequenceEqual(b.signature);
+			}
+
+			public static bool operator !=(StratisBlockSignature a, StratisBlockSignature b)
+			{
+				return !(a == b);
+			}
+
+			#region IBitcoinSerializable Members
+
+			public void ReadWrite(BitcoinStream stream)
+			{
+				stream.ReadWriteAsVarString(ref signature);
+			}
+
+			#endregion
+
+			public override string ToString()
+			{
+				return Encoders.Hex.EncodeData(this.signature);
```

Comment:
This can throw an `ArgumentNullException` when `this.signature` is `null`. What should `ToString` return in case `signature` is empty?